### ISSUE TITLE
torcs,nvidia -- libGL error: unable to load driver: swrast_dri.so

### ISSUE THREAD

**Issue by wizzup** [NixOS/nixpkgs#30749]
## Issue description
my `torcs` building issue (#30680) was fixed (#30696) but when I try running `torcs` I get following error:

    libGL error: unable to load driver: swrast_dri.so
    libGL error: failed to load driver: swrast

running with `LIBGL_DEBUG=verbose` gives

    [nix-shell:~]$ LIBGL_DEBUG=verbose torcs 
    libGL: screen 0 does not appear to be DRI2 capable
    libGL: OpenDriver: trying /run/opengl-driver/lib/dri/tls/swrast_dri.so
    libGL: OpenDriver: trying /run/opengl-driver/lib/dri/swrast_dri.so
    libGL: dlopen /run/opengl-driver/lib/dri/swrast_dri.so failed (/run/opengl-driver/lib/dri/swrast_dri.so: cannot open shared object file: No such file or directory)
    libGL error: unable to load driver: swrast_dri.so
    libGL error: failed to load driver: swrast
    Visual Properties Report
    ------------------------
    Compatibility mode, properties unknown.
    freeglut (/nix/store/rrbq4m95d0rmf07p2hnh8709i65jcrpb-torcs-1.3.7/lib/torcs/torcs-bin):  ERROR:  Internal error <FBConfig with necessary capabilities not found> in function fgOpenWindow

I use following settings in my `configuration.nix`:

    hardware.opengl = {
      driSupport = true;
      driSupport32Bit = true;
    };

    services.xserver.videoDrivers = [ "nvidia" ];

I have no problem running `glxinfo` and `glxgears`.

### Steps to reproduce

    $ nix-shell -I nixpkgs=/nix/nixpkgs -p torcs

    [nix-shell:~]$ LIBGL_DEBUG=verbose torcs 

## Technical details

 * System: NixOS: 18.03.git.00e56fb (Impala)
 * Nix version: nix-env (Nix) 1.11.15
 * Nixpkgs version: "18.03pre117327.3fe7cddc30"
 * Sandboxing enabled: build-use-sandbox = true

**Comment by vcunat** [comment]
The wrapping with mesa path in the expression only makes things worse.  I would certainly remove it.  The OS is who should be adding libGL into LD_LIBRARY_PATH.

**Comment by bugaev** [comment]
I have exactly same run-time error message while running nixos.asymptote as I described in this [issue](https://github.com/NixOS/nixpkgs/issues/31097):

    libGL error: unable to load driver: swrast_dri.so
    libGL error: failed to load driver: swrast
    freeglut (asy):  ERROR:  Internal error <FBConfig with necessary capabilities not found> in function fgOpenWindow

I am using nvidia drivers.

**Comment by StijnDW** [comment]
I might have the same issue with the package `factor-lang`
I'm also using NVidia drivers.

Program output:
```
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast
The program 'factor' received an X Window System error.
This probably reflects a bug in the program.
The error was 'GLXBadContext'.
  (Details: serial 227 error_code 169 request_code 154 minor_code 6)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the --sync command line
   option to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
```

My system:
- System: NixOS 17.09.1989.a86c8599a4 (Hummingbird)
- Nix version: nix-env (Nix) 1.11.15
- Nixpkgs version: "17.09.1989.a86c8599a4"
- Sandboxing enabled: false

**Comment by vcunat** [comment]
Yes, that's the same packaging mistake.

**Comment by vcunat** [comment]
@wizzup: torcs should be fixed now, and on 17.09 as well.

**Comment by vcunat** [comment]
@StijnDW: fixed factor-lang on master and 17.09.  *If you use channels, those may take a couple days to catch up.*

We've solved the `swrast` lines for asymptote on the other thread, so let me close this ticket.
